micromagneticmodel/abstract/container.py

```diff
@@ -328,10 +322,17 @@ def __add__(self, other):
             result._terms.append(term)
 
         if isinstance(other, self._term_class):
-            if other in result:
-                msg = f"Cannot have two {other.__class__} terms in the container."
+            if any(
+                isinstance(other, term.__class__) and other.name == term.name
+                for term in self._terms
+            ):
+                msg = (
+                    f"Cannot have same class {other.__class__} and name {other.name} "
```

Comment:
I would add a sentence that there already is a term of the same type with `other.name` in the container. I think that is not made clear enough.

Reply:
Let me know what you think after the changes.

Reply:
The changed error message is very clear.